### DISCUSSION TITLE
panic: runtime error: slice bounds out of range

### DISCUSSION THREAD

**Discussion by zeales-go** [sijms/go-ora#90] [Q&A] [answered]
CREATE TABLE ttt (
  srvid                          NUMBER(16)       NOT NULL,
  itype                          VARCHAR2(4)      NOT NULL,
  srvname                        VARCHAR2(255)    NOT NULL,
  supperid                       VARCHAR2(2)      NOT NULL,
  status                         VARCHAR2(1)      NOT NULL,
  memo                           VARCHAR2(255)        NULL,
  stdfee                         NUMBER(12,2)         NULL,
  areas                          VARCHAR2(1024)       NULL,
  definition                     VARCHAR2(1)          NULL,
  inputway                       VARCHAR2(8)          NULL
);

insert into ttt (SRVID, ITYPE, SRVNAME, SUPPERID, STATUS, MEMO, STDFEE, AREAS, DEFINITION, INPUTWAY)
values (100, 'IPTV', 'ttt', 'J', 'Y', 'mytest', null, '90,99,100,101,102,104,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,272,273,274,598,599,600,601,602,603,604,605,606,607,608,609,610,611,612,613,614,615,616,617,618,619,620,621,622,623,624,625,626,627,628,629,630,631,632,633,634,635,636,637,638,640,641,642,643,644,645,646,647,648,649,650,651,653,654,655,656,657,658,659,660,661,665,666,667,670,671,672,673,674,680,681,682,683,684,685,686,687,690,691,693,694,695,696,715,716,717,718,719,720,721,722,730,750,751,752,753,754,755,756,757,758,759,760,761,762,815,816,817,818,820,821,822,823,900,901,902,903,904,915,916,917,920,921,922,923,924,925,930,931,932,106,668', null, null);

when: select * from ttt where srvid=100 
panic: runtime error: slice bounds out of range [-119:]

goroutine 1 [running]:
github.com/sijms/go-ora/v2/network.(*Session).GetInt64(0xc00009e280, 0x7f, 0x101, 0x363, 0x0, 0x0)
	D:/Boss/gopackage/src/github.com/sijms/go-ora/v2/network/session.go:771 +0x223
github.com/sijms/go-ora/v2/network.(*Session).GetInt(0xc00009e280, 0x1, 0x101, 0x0, 0x36a, 0x0)
	D:/Boss/gopackage/src/github.com/sijms/go-ora/v2/network/session.go:784 +0x52
github.com/sijms/go-ora/v2/network.(*Session).GetClr(0xc00009e280, 0x0, 0x0, 0x0, 0x0, 0x0)
	D:/Boss/gopackage/src/github.com/sijms/go-ora/v2/network/session.go:833 +0x1df
github.com/sijms/go-ora/v2.(*defaultStmt).read(0xc0000cc820, 0xc000098500, 0x0, 0x0)
	D:/Boss/gopackage/src/github.com/sijms/go-ora/v2/command.go:596 +0x55f
github.com/sijms/go-ora/v2.(*Stmt).Query(0xc0000cc820, 0xf248d8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	D:/Boss/gopackage/src/github.com/sijms/go-ora/v2/command.go:1240 +0x41f
database/sql.ctxDriverStmtQuery(0x8379c0, 0xc00000a108, 0x837b80, 0xc0000cc820, 0xf248d8, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	C:/golang1.14/src/database/sql/ctxutil.go:94 +0x16c
database/sql.rowsiFromStatement(0x8379c0, 0xc00000a108, 0x837080, 0xc0000ca000, 0xc0000b0800, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	C:/golang1.14/src/database/sql/sql.go:2624 +0x171

the trace file was:
Write packet:
00000000  00 15 00 00 06 00 00 00  00 00 11 69 00 01 01 01  |...........i....|
00000010  01 02 03 93 00                                    |.....|
2021-08-10T09:11:33.8944: 
Read packet:
00000000  00 0f 00 00 06 00 00 00  00 00 09 01 01 01 0d     |...............|
2021-08-10T09:11:33.8954: Prepare
select * from swp_service_info where srvid=100
2021-08-10T09:11:33.8954: Query:
select * from swp_service_info where srvid=100
2021-08-10T09:11:33.8954: 
Write packet:
00000000  00 73 00 00 06 00 00 00  00 00 03 5e 00 02 80 21  |.s.........^...!|
00000010  00 01 01 2e 01 01 0d 00  00 00 01 19 04 7f ff ff  |................|
00000020  ff 00 00 00 00 00 00 00  00 00 00 00 01 00 00 00  |................|
00000030  00 00 2e 73 65 6c 65 63  74 20 2a 20 66 72 6f 6d  |...select * from|
00000040  20 73 77 70 5f 73 65 72  76 69 63 65 5f 69 6e 66  | swp_service_inf|
00000050  6f 20 77 68 65 72 65 20  73 72 76 69 64 3d 31 30  |o where srvid=10|
00000060  30 01 01 01 19 00 00 00  00 00 01 01 00 02 80 00  |0...............|
00000070  00 00 00                                          |...|
2021-08-10T09:11:33.9050: 
Read packet:
00000000  04 c1 00 00 06 00 00 00  00 00 10 17 6c ff bd 56  |............l..V|
00000010  76 ae 78 7f 9a b4 46 57  4a 38 24 1b 78 79 08 0a  |v.x...FWJ8$.xy..|
00000020  0a 05 31 02 06 3a 01 0a  51 02 00 10 00 01 16 00  |..1..:..Q.......|
00000030  00 00 00 00 00 00 00 05  01 05 05 53 52 56 49 44  |...........SRVID|
00000040  00 00 00 00 01 80 00 00  01 04 00 00 00 00 02 03  |................|
00000050  54 01 01 04 00 05 01 05  05 49 54 59 50 45 00 00  |T........ITYPE..|
00000060  01 01 00 01 80 00 00 01  ff 00 00 00 00 02 03 54  |...............T|
00000070  01 01 ff 00 07 01 07 07  53 52 56 4e 41 4d 45 00  |........SRVNAME.|
00000080  00 01 02 00 01 80 00 00  01 02 00 00 00 00 02 03  |................|
00000090  54 01 01 02 00 08 01 08  08 53 55 50 50 45 52 49  |T........SUPPERI|
000000a0  44 00 00 01 03 00 01 80  00 00 01 01 00 00 00 00  |D...............|
000000b0  02 03 54 01 01 01 00 06  01 06 06 53 54 41 54 55  |..T........STATU|
000000c0  53 00 00 01 04 00 01 80  00 00 01 ff 00 00 00 00  |S...............|
000000d0  02 03 54 01 01 ff 01 04  01 04 04 4d 45 4d 4f 00  |..T........MEMO.|
000000e0  00 01 05 00 02 00 0c 01  02 01 16 00 00 00 00 00  |................|
000000f0  00 00 01 06 01 06 06 53  54 44 46 45 45 00 00 01  |.......STDFEE...|
00000100  06 00 01 80 00 00 02 04  00 00 00 00 00 02 03 54  |...............T|
00000110  01 02 04 00 01 05 01 05  05 41 52 45 41 53 00 00  |.........AREAS..|
00000120  01 07 00 01 80 00 00 01  01 00 00 00 00 02 03 54  |...............T|
00000130  01 01 01 01 0a 01 0a 0a  44 45 46 49 4e 49 54 49  |........DEFINITI|
00000140  4f 4e 00 00 01 08 00 01  80 00 00 01 08 00 00 00  |ON..............|
00000150  00 02 03 54 01 01 08 01  08 01 08 08 49 4e 50 55  |...T........INPU|
00000160  54 57 41 59 00 00 01 09  00 01 07 07 78 79 08 0a  |TWAY........xy..|
00000170  0a 0c 28 01 01 02 1f e8  01 02 01 02 00 06 22 01  |..(...........".|
00000180  0a 00 01 32 00 00 00 07  02 c2 02 04 49 50 54 56  |...2........IPTV|
00000190  0a b1 ea c7 e5 bb f9 b1  be b0 fc 01 4a 01 59 06  |............J.Y.|
000001a0  6d 79 74 65 73 74 00 fe  ff 39 30 2c 39 39 2c 31  |mytest...90,99,1|
000001b0  30 30 2c 31 30 31 2c 31  30 32 2c 31 30 34 2c 32  |00,101,102,104,2|
000001c0  30 30 2c 32 30 31 2c 32  30 32 2c 32 30 33 2c 32  |00,201,202,203,2|
000001d0  30 34 2c 32 30 35 2c 32  30 36 2c 32 30 37 2c 32  |04,205,206,207,2|
000001e0  30 38 2c 32 30 39 2c 32  31 30 2c 32 31 31 2c 32  |08,209,210,211,2|
000001f0  31 32 2c 32 31 33 2c 32  31 34 2c 32 31 35 2c 32  |12,213,214,215,2|
00000200  31 36 2c 32 31 37 2c 32  31 38 2c 32 31 39 2c 32  |16,217,218,219,2|
00000210  32 30 2c 32 32 31 2c 32  32 32 2c 32 32 33 2c 32  |20,221,222,223,2|
00000220  32 34 2c 32 32 35 2c 32  32 36 2c 32 32 37 2c 32  |24,225,226,227,2|
00000230  32 38 2c 32 32 39 2c 32  33 30 2c 32 33 31 2c 32  |28,229,230,231,2|
00000240  33 32 2c 32 33 33 2c 32  33 34 2c 32 37 32 2c 32  |32,233,234,272,2|
00000250  37 33 2c 32 37 34 2c 35  39 38 2c 35 39 39 2c 36  |73,274,598,599,6|
00000260  30 30 2c 36 30 31 2c 36  30 32 2c 36 30 33 2c 36  |00,601,602,603,6|
00000270  30 34 2c 36 30 35 2c 36  30 36 2c 36 30 37 2c 36  |04,605,606,607,6|
00000280  30 38 2c 36 30 39 2c 36  31 30 2c 36 31 31 2c 36  |08,609,610,611,6|
00000290  31 32 2c 36 31 33 2c 36  31 34 2c 36 31 35 2c 36  |12,613,614,615,6|
000002a0  31 36 2c 36 31 37 2c 36  ff 31 38 2c 36 31 39 2c  |16,617,6.18,619,|
000002b0  36 32 30 2c 36 32 31 2c  36 32 32 2c 36 32 33 2c  |620,621,622,623,|
000002c0  36 32 34 2c 36 32 35 2c  36 32 36 2c 36 32 37 2c  |624,625,626,627,|
000002d0  36 32 38 2c 36 32 39 2c  36 33 30 2c 36 33 31 2c  |628,629,630,631,|
000002e0  36 33 32 2c 36 33 33 2c  36 33 34 2c 36 33 35 2c  |632,633,634,635,|
000002f0  36 33 36 2c 36 33 37 2c  36 33 38 2c 36 34 30 2c  |636,637,638,640,|
00000300  36 34 31 2c 36 34 32 2c  36 34 33 2c 36 34 34 2c  |641,642,643,644,|
00000310  36 34 35 2c 36 34 36 2c  36 34 37 2c 36 34 38 2c  |645,646,647,648,|
00000320  36 34 39 2c 36 35 30 2c  36 35 31 2c 36 35 33 2c  |649,650,651,653,|
00000330  36 35 34 2c 36 35 35 2c  36 35 36 2c 36 35 37 2c  |654,655,656,657,|
00000340  36 35 38 2c 36 35 39 2c  36 36 30 2c 36 36 31 2c  |658,659,660,661,|
00000350  36 36 35 2c 36 36 36 2c  36 36 37 2c 36 37 30 2c  |665,666,667,670,|
00000360  36 37 31 2c 36 37 32 2c  36 37 33 2c 36 37 34 2c  |671,672,673,674,|
00000370  36 38 30 2c 36 38 31 2c  36 38 32 2c 36 38 33 2c  |680,681,682,683,|
00000380  36 38 34 2c 36 38 35 2c  36 38 36 2c 36 38 37 2c  |684,685,686,687,|
00000390  36 39 30 2c 36 39 31 2c  36 39 33 2c 36 39 34 2c  |690,691,693,694,|
000003a0  36 39 35 2c 36 39 36 2c  c3 37 31 35 2c 37 31 36  |695,696,.715,716|
000003b0  2c 37 31 37 2c 37 31 38  2c 37 31 39 2c 37 32 30  |,717,718,719,720|
000003c0  2c 37 32 31 2c 37 32 32  2c 37 33 30 2c 37 35 30  |,721,722,730,750|
000003d0  2c 37 35 31 2c 37 35 32  2c 37 35 33 2c 37 35 34  |,751,752,753,754|
000003e0  2c 37 35 35 2c 37 35 36  2c 37 35 37 2c 37 35 38  |,755,756,757,758|
000003f0  2c 37 35 39 2c 37 36 30  2c 37 36 31 2c 37 36 32  |,759,760,761,762|
00000400  2c 38 31 35 2c 38 31 36  2c 38 31 37 2c 38 31 38  |,815,816,817,818|
00000410  2c 38 32 30 2c 38 32 31  2c 38 32 32 2c 38 32 33  |,820,821,822,823|
00000420  2c 39 30 30 2c 39 30 31  2c 39 30 32 2c 39 30 33  |,900,901,902,903|
00000430  2c 39 30 34 2c 39 31 35  2c 39 31 36 2c 39 31 37  |,904,915,916,917|
00000440  2c 39 32 30 2c 39 32 31  2c 39 32 32 2c 39 32 33  |,920,921,922,923|
00000450  2c 39 32 34 2c 39 32 35  2c 39 33 30 2c 39 33 31  |,924,925,930,931|
00000460  2c 39 33 32 2c 31 30 36  2c 36 36 38 00 00 00 08  |,932,106,668....|
00000470  01 06 04 8f 7d ec 37 02  0d 45 01 01 00 00 00 00  |....}.7..E......|
00000480  00 00 04 01 01 01 0e 01  01 02 05 7b 00 00 01 01  |...........{....|
00000490  01 0e 03 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000004a0  00 01 01 00 00 00 00 19  4f 52 41 2d 30 31 34 30  |........ORA-0140|
000004b0  33 3a 20 6e 6f 20 64 61  74 61 20 66 6f 75 6e 64  |3: no data found|
000004c0  0a                                                |.|

**Comment by sijms** [comment, marked as answer]
what version you are using?
are you using code like this or this is for demonstration:
```select * from ttt where srvid=100```
as you need to specify column name in sql

**Reply by zeales-go** [reply]
V2, when column length greater then 254， this problem will happened。

e.g: select lpad('a', 254, '0') as aa from dual 。

**Reply by zeales-go** [reply]
when useing go-ora （not v2），everything done well，thanks for your contribution，i can run without cgo+oci now。thank you very much。
